stdlib/public/runtime/SwiftRT-ELF-WASM.cpp

```diff
@@ -20,11 +21,27 @@
 
 #if defined(__ELF__)
 extern "C" const char __dso_handle[];
+extern "C" const char __ehdr_start[] __attribute__((__weak__));
+#endif
+
+static const void *getBaseAddress(void) {
+#if defined(__ELF__)
+  if (&__ehdr_start != nullptr) {
+    return __ehdr_start;
+  }
+
+  // We've hit an edge case where the linker was invoked such that the ELF
+  // header was not placed in a loadable section (or some such) and as a result
+  // __ehdr_start was not declared. Fall back to calling dladdr().
+  if (auto info = swift::SymbolInfo::lookup(__dso_handle)) {
```

Comment:
Ironically _this_ is what failed to link.

Reply:
(Because this file is compiled to an object file and `SymbolInfo` of course isn't exported. I'll need to call `dladdr()` directly instead.)